### PR TITLE
feat: always pass the repo owner to the credential helper

### PR DESCRIPTION
**Pull request by garethjevans**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

**Comment by abayer** [comment]
Mind creating/linking with `fixes #...` an issue? =)

**Comment by garethjevans** [comment]
fixes #6722

File: pkg/cmd/step/git/credentials/step_git_credentials.go [hunk 1]
```diff
@@ -25,16 +25,20 @@ import (
 )
 
 const (
-	optionOutputFile     = "output"
+	optionOutputFile = "output"
+	// Deprecated
```
Comment (hferentschik) [inline pkg/cmd/step/git/credentials/step_git_credentials.go]: What would be our plan to eventually remove this? Should we at least create an issue to keep of track and maybe add a 'deprecation-followup' label of similar? One could then on a regular basis review these issues and evaluate of a certain deprecated feature can be removed. 

I think it is pretty much only the SaaS which uses this, so provided you can update in one go, I'd even be tempted to remove directly. WDYT?

File: pkg/cmd/step/verify/step_verify_environments.go [hunk 2]
```diff
@@ -510,9 +510,9 @@ func (o *StepVerifyEnvironmentsOptions) pushDevEnvironmentUpdates(environmentRep
 
 	err = gitter.Push(localRepoDir, remoteName, true, "master")
 	if err != nil {
-		return errors.Wrapf(err, "unable to push %s to %s", localRepoDir, environmentRepo.URL)
+		return errors.Wrapf(err, "unable to push %s to %s", localRepoDir, environmentRepo.CloneURL)
 	}
-	log.Logger().Infof("Pushed Git repository to %s\n\n", util.ColorInfo(environmentRepo.HTMLURL))
+	log.Logger().Infof("Pushed Git repository to %s", util.ColorInfo(environmentRepo.HTMLURL))
```
Comment (hferentschik) [inline pkg/cmd/step/verify/step_verify_environments.go]: 👍

File: pkg/gits/helpers.go [hunk 3]
```diff
@@ -713,13 +716,15 @@ func ForkAndPullRepo(gitURL string, dir string, baseRef string, branchName strin
...
 
-func configureJxAsGitCredentialHelper(dir string, gitter Gitter) error {
+func configureJxAsGitCredentialHelper(dir string, gitter Gitter, repoOwner string) error {
 	// configure jx as git credential helper for this repo
 	jxProcessBinary, err := os.Executable()
 	if err != nil {
 		return errors.Wrapf(err, "unable to determine jx binary location")
 	}
-	return gitter.Config(dir, "--local", "credential.helper", fmt.Sprintf("%s step git credentials --credential-helper", jxProcessBinary))
+	config := []string{"--local", "credential.helper", fmt.Sprintf("%s step git credentials --credential-helper --repo-owner %s", jxProcessBinary, repoOwner)}
```
Comment (hferentschik) [inline pkg/gits/helpers.go]: +1, definitely feels like the smallest possible change to make this work for your use case which at the same time is not completely irrational for the other users.

File: pkg/cmd/step/git/credentials/step_git_credentials.go (hunk 1 above)
Comment (garethjevans) [reply pkg/cmd/step/git/credentials/step_git_credentials.go]: This command (unfortunately) is embedded elsewhere in the SaaS pipelines so I’m not going to be able to remove it in one go, but should be able to in a few weeks once we’ve migrated over. I’ll create a task to do that

Comment (garethjevans) [reply pkg/cmd/step/git/credentials/step_git_credentials.go]: Taks to remove #6723

**Pull request merged**